### ISSUE TITLE
`RigidBody::setIsActive` / `RigidBody::setIsSleeping` behaviour

### ISSUE THREAD

**Issue by Casqade** [DanielChappuis/reactphysics3d#284]
I'm slightly confused with sleeping/active body state concepts in the context of their implementation. 
e.g. `RigidBody::setIsActive` should either: 
- Put the body to sleep (via `RigidBody::setIsSleeping`), then deactivate it
- Or awake the body, then activate it

But both the code and the comment in `RigidBody::setIsSleeping` suggest that "If the body is not active, do nothing (it is sleeping)". That means when we try to activate the body with `body->setIsActive(true)`, the simplified call flow would be as follows:
1. `RigidBody::setIsSleeping(false)` (RigidBody.cpp:991 - body is not yet active, so the code returns without awaking it) 
2. `CollisionBody::setIsActive(true)` (finally activate the body)

In result, our body has been activated, but not awaken. Is it an intended behaviour or I'm missing something ?

**Comment by DanielChappuis** [comment]
Hello,

Sorry for my late answer. The idea is that a body is "active" when it is simulated by the engine (the engine will do something with the body) and "not active" when it is not simulated. On the other side, a body is "sleeping" when it simulated (active) but is not moving for some time and has no forces or constraints on it. Therefore, it's an engine optimization not to compute things for 'sleeping' bodies.

Therefore, a body should be deactivated when you do want the physics engine to updated anything with your body (it's like removing it from the world for some time). This feature is probably not used a lot.

Putting a body to sleep of waking it up is automatically done by the physics engine most of the time. You should not have to do it yourself but it happens that you want to awake a body that is sleeping a some point. To do that you can wake up a body using the `RigidBody::setIsSleeping(false)` method.

I hope that it is more clear.

**Comment by Casqade** [comment]
> I hope that it is more clear.

Well, it is clear from logical point of view, but my question is more related to implementation details. I expect `RigidBody::setIsActive(true)` to automatically wake up the body. However, that is not the case:

```
rp3d::RigidBody* body = world.createRigidBody({});
body->setIsActive(false);
assert( body->isSleeping() == true ); <-- ok

body->setIsActive(true);
assert( body->isSleeping() == false ); <-- assertion failure
```

In my initial post I described why it is happening, but I'm still not sure whether this is the intended behaviour. Looks like a bug or 'logical inconsistency' to me, actually...

**Comment by Casqade** [comment]
> the engine will check, the next frame, if the body is still sleeping or not

If you're completely sure it works as described, then I don't have any problems with it. The confusion arose from [this](https://github.com/DanielChappuis/reactphysics3d/issues/235) (probably unfixed?) issue and my own experience.

**Comment by DanielChappuis** [comment]
Yes it is the theory of how it should work at least and yes the issue #235 is not fixed yet.

**Comment by DanielChappuis** [comment]
The issue #235 is now fixed in version [v0.10.0](https://github.com/DanielChappuis/reactphysics3d/releases/tag/v0.10.0) of the library. Thanks a lot for reporting the issue.
